packages/core/scripts/remove-collateral-upp/1_Propose.js

```diff
@@ -0,0 +1,117 @@
+
+const AddressWhitelist = getTruffleContract("AddressWhitelist", web3, "latest");
+const Finder = getTruffleContract("Finder", web3, "latest");
+const Governor = getTruffleContract("Governor", web3, "latest");
+const Voting = getTruffleContract("Voting", web3, "latest");
+
+const { interfaceName } = require("@uma/common");
+const { GasEstimator } = require("@uma/financial-templates-lib");
+
+const _ = require("lodash");
+const winston = require("winston");
+
+const argv = require("minimist")(process.argv.slice(), { string: ["collateral"] });
+
+const proposerWallet = "0x2bAaA41d155ad8a4126184950B31F50A1513cE25";
+
+async function runExport() {
+  console.log("Running Upgrade🔥");
+
+  const netId = await web3.eth.net.getId();
+  console.log("Connected to network id", netId);
+
+  const gasEstimator = new GasEstimator(
+    winston.createLogger({ silent: true }),
+    60, // Time between updates.
+    netId
+  );
+
+  if (!argv.collateral) {
+    throw new Error("Must provide --collateral");
+  }
+
+  const collaterals = _.castArray(argv.collateral);
+
+  const argObjects = _.zipWith(collaterals, (collateral) => {
+    return { collateral };
+  });
+
+  const getTxns = async ({ collateral }) => {
+    console.log("Examining collateral", collateral);
+
+    const txns = [];
+
+    // The proposal will remove the currency from the whitelist.
+    const whitelist = await AddressWhitelist.deployed();
+    if (!(await whitelist.isOnWhitelist(collateral))) {
+      throw new Error("Collateral not on whitelist");
+    } else {
+      console.log("Collateral", collateral, "is on the whitelist. Removing it.");
+      const removeCollateralFromWhitelistTx = whitelist.contract.methods.removeFromWhitelist(collateral).encodeABI();
+      console.log("removeCollateralFromWhitelistTx", removeCollateralFromWhitelistTx);
+      txns.push({ to: whitelist.address, value: 0, data: removeCollateralFromWhitelistTx });
+
+      console.log(`
+
+      Collateral currency to remove: ${collateral}
+
+      `);
+    }
+
+    return txns;
+  };
+
+  let transactionList = [];
+  for (let argObject of argObjects) {
+    const transactionsToAdd = await getTxns(argObject);
+    transactionList = [...transactionList, ...transactionsToAdd];
+  }
+
+  const governor = await Governor.deployed();
+  console.log(`Sending to governor @ ${governor.address}`);
+
+  // Send the proposal
+  await gasEstimator.update();
+  const txn = await governor.propose(transactionList, {
+    from: proposerWallet,
+    gasPrice: gasEstimator.getCurrentFastPrice(),
+  });
+  console.log("Transaction: ", txn?.tx);
+
+  const finder = await Finder.deployed();
+  const oracleAddress = await finder.getImplementationAddress(web3.utils.utf8ToHex(interfaceName.Oracle));
+  console.log(`Governor submitting admin request to Voting @ ${oracleAddress}`);
+
+  const oracle = await Voting.deployed();
+  const priceRequests = await oracle.getPastEvents("PriceRequestAdded");
```

Comment:
I think you should optimize this `getPastEvents` call to just the `txn.block` since that's the only event we care about, right?